chapter_linear-networks/linear-regression-scratch.md

```diff
@@ -68,28 +68,15 @@ To make our problem easy, we will set its standard deviation to 0.01.
 The following code generates our synthetic dataset.
 
 ```{.python .input}
+#@tab mxnet, pytorch
 def synthetic_data(w, b, num_examples):  #@save
     """Generate y = Xw + b + noise."""
-    X = np.random.normal(0, 1, (num_examples, len(w)))
-    y = np.dot(X, w) + b
-    y += np.random.normal(0, 0.01, y.shape)
+    X = d2l.normal(0, 1, (num_examples, len(w)))
+    y = d2l.matmul(X, w) + b
```

Comment:
why not enable d2l.matmul for TF code?

Reply:
i see d2l.matmul is enabled for TF in https://github.com/d2l-ai/d2l-en/pull/1182/files
However, why couldn't we enable d2l.normal for TF? Is it due to limitations in TF? @terrytangyuan